dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/rest/TelemetryClientFilter.java

```diff
@@ -51,6 +51,19 @@ public class TelemetryClientFilter implements ClientRequestFilter, ClientRespons
     private static final NetClientAttributesGetterImpl NET_CLIENT_ATTRIBUTES_GETTER = new NetClientAttributesGetterImpl();
     private static final HttpClientAttributesGetterImpl HTTP_CLIENT_ATTRIBUTES_GETTER = new HttpClientAttributesGetterImpl();
 
+    /**
+     * Retrieve the TelemetryClientFilter for the current application using CDI
+     * <p>
+     * Implementation note: It's important that there's a class which is registered as a CDI bean on the stack from this bundle when {@code CDI.current()} is called.
```

Comment:
I assume that you mean an _application_ class? What happens if there is not? I would imagine that in that case this code wouldn't be needed in the first place ... so why the note? I must be missing something.

Reply:
When we call `CDI.current()` CDI will do a stack walk to find the current BDA. The correct answer is the BDA for this bundle which provides the telemetry classes.

For it to do this, we must have a class which CDI knows about from this BDA on the stack. IIRC, when we use `CDIExtensionMetadata` the BDA that gets generated is only aware of the classes we return as bean classes.

If there isn't such a class on the stack (which I think would be the case if we called `CDI.current()` from `TelemetryClientBuilderListener`), CDI will continue up the stack to find the BDA to use, which may not be the correct one. Best case it's an application BDA which can see `TelemetryClientFilter`. Worst case it's a different extension BDA which can't see it.

Reply:
So this method is broken under some circumstances?